File: src/main/java/app/pdfx/MdStruct.java
package app.pdfx;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)

public @interface MdStruct {
    String name() default "";

    enum StructType {
        MD_STRUCT,
        MD_ENABLE_STRUCT,
    }

    StructType type() default StructType.MD_STRUCT;

    enum Access {
        READ_ONLY,
        READ_WRITE,
    }

    Access access() default Access.READ_WRITE;
}

File: src/main/kotlin/app/pdfx/MdStruct.kt
package app.pdfx

@Retention(AnnotationRetention.RUNTIME)
annotation class MdStruct(
    val name: String = "",
    val type: StructType = StructType.MD_STRUCT,
    val access: Access = Access.READ_WRITE
) {
    enum class StructType {
        MD_STRUCT,
        MD_ENABLE_STRUCT
    }

    enum class Access {
        READ_ONLY,
        READ_WRITE
    }
}
